boards/arm/xenvm/doc/index.rst

```diff
@@ -0,0 +1,171 @@
+.. xenvm:
+
+ARMv8 Xen Virtual Machine
+#########################
+
+Overview
+********
+
+This board allows to run Zephyr as Xen guest on any ARMv8 board that supports
+ARM Virtualization Extensions.
+
+.. figure:: xen_project_logo.png
+   :align: center
+   :alt: XenVM
+
+   Xen virtual Guest (Credit: Xen Project)
+
+It provides minimal set of devices:
+
+* ARM Generic timer
+* GICv2
+* SBSA (subset of PL011) UART controller
+
+
+Hardware
+********
+Supported Features
+==================
+
+The following hardware features are supported:
+
++--------------+-------------+----------------------+
+| Interface    | Controller  | Driver/Component     |
++==============+=============+======================+
+| GIC          | virtualized | interrupt controller |
++--------------+-------------+----------------------+
+| SBSA UART    | emulated    | serial port          |
++--------------+-------------+----------------------+
+| ARM TIMER    | virtualized | system clock         |
++--------------+-------------+----------------------+
+
+The kernel currently does not support other hardware features on this platform.
+
+Devices
+========
+System Clock
+------------
+
+This board configuration uses a system clock frequency of 8.32 MHz. This is the
+default value, which should be corrected for user's actual hardware.
+
+You can determine clock frequency of your ARM Generic Timer by inspecting Xen
+boot log:
+
+::
+
+  (XEN) [    0.147541] Generic Timer IRQ: phys=30 hyp=26 virt=27 Freq: 8320 KHz
+
+Serial Port
+-----------
+
+This board configuration uses a single serial communication channel using SBSA
+UART. This is a minimal UART implementation provided by Xen. Xen PV Console is
+not supported at this moment.
+
+Interrupt Controller
+--------------------
+
+By default, GICv2 is selected. If your hardware is based on GICv3, you can
+configure Zephyr to use it, by amending device tree and Kconfig
+option as well as guest configuration file.
+
+Known Problems or Limitations
+==============================
+
+Xen configures guests in runtime by providing device tree that describes guest
+environment. On other hand, Zephyr uses static configuration that should be know
+at build time. So there are chances, that Zephyr image created with default
+configuration would not boot on your hardware. In this case you need to update
+configuration by altering device tree and Kconfig options. This will be covered
+in detail in next section.
+
+No Xen-specific features are supported at the moment. This includes:
+
+* Xen Enlighten memore page
```

Comment:
I might not follow up, is `memore` right?

Reply:
Yes, thanks for the pointing out